pkg/api/v1beta3/dynakube/validation/logmodule.go

```diff
@@ -0,0 +1,48 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	errorConflictingLogModule = "The DynaKube's specification tries to enable LogModule in a namespace where another DynaKube already deploys the OneAgent, which is not supported. The conflicting DynaKube: %s"
+
+	errorConflictingOneAgentSpec = "The DynaKube's specification tries to enable LogModule and OneAgent at the same time, which is not supported. Please disable the LogModule or OneAgent"
+)
+
+func conflictingLogModuleNodeSelector(ctx context.Context, dv *Validator, dk *dynakube.DynaKube) string {
+	if !dk.Spec.LogModule.Enabled {
+		return ""
+	}
+
+	if dk.NeedsOneAgent() {
+		return errorConflictingOneAgentSpec
+	}
+
+	validDynakubes := &dynakube.DynaKubeList{}
+	if err := dv.apiReader.List(ctx, validDynakubes, &client.ListOptions{Namespace: dk.Namespace}); err != nil {
+		log.Info("error occurred while listing dynakubes", "err", err.Error())
+
+		return ""
+	}
+
+	for _, item := range validDynakubes.Items {
+		if item.Name == dk.Name {
+			continue
+		}
+
+		if item.NeedsOneAgent() {
+			if hasConflictingMatchLabels(dk.LogModuleNodeSelector(), item.OneAgentNodeSelector()) {
+				log.Info("requested dynakube has conflicting LogModule nodeSelector", "name", dk.Name, "namespace", dk.Namespace)
+
+				return fmt.Sprintf(errorConflictingLogModule, item.Name)
+			}
+		}
+	}
```

Comment:
here you could also collect all the conflicting DynaKubes 😅

Reply:
just make sure to only list the conflicting dynakubes. if we output the full error for each dynakube we could easily hit the warning limit

Reply:
We don't need that, as there can only be 1 conflicting dynakube.

Reply:
I could be many, but it's not very realistic.

Example
```
apiVersion: dynatrace.com/v1beta2
kind: DynaKube
metadata:
  name: cloudnative
  namespace: dynatrace
spec:
  apiUrl: https://zib50933.dev.dynatracelabs.com/api
  tokens: zib50933
  enableIstio: true
  dynatraceApiRequestThreshold: 15

  oneAgent:
    cloudNativeFullStack:
      namespaceSelector:
        matchLabels:
          monitored/by: v2
      nodeSelector:
          dk-v2: deploy
      tolerations:
       - effect: NoSchedule
         key: node-role.kubernetes.io/master
         operator: Exists
       - effect: NoSchedule
         key: node-role.kubernetes.io/control-plane
         operator: Exists
---
apiVersion: dynatrace.com/v1beta1
kind: DynaKube
metadata:
  name: cloudnative-old
  namespace: dynatrace
spec:
  apiUrl: https://zib50933.dev.dynatracelabs.com/api
  tokens: zib50933

  namespaceSelector:
     matchLabels:
       monitored/by: v1

  oneAgent:
   cloudNativeFullStack:
     nodeSelector:
      dk-v1: deploy
     tolerations:
       - effect: NoSchedule
         key: node-role.kubernetes.io/master
         operator: Exists
       - effect: NoSchedule
         key: node-role.kubernetes.io/control-plane
         operator: Exists
---
apiVersion: dynatrace.com/v1beta3
kind: DynaKube
metadata:
  name: log-module
  namespace: dynatrace
spec:
  apiUrl: https://zib50933.dev.dynatracelabs.com/api
  tokens: zib50933
  dynatraceApiRequestThreshold: 15
  logModule:
    enabled: true
```

Reply:
😮  - That's good, an edge case I didn't think of, will adapt the code and add it as unit test. Thx

Reply:
Done